raft/src/main/java/org/apache/kafka/raft/QuorumState.java

```diff
@@ -196,7 +203,15 @@ public void initialize(OffsetAndEpoch logEndOffsetAndEpoch) throws IOException,
     }
 
     public Set<Integer> remoteVoters() {
-        return voters.stream().filter(voterId -> voterId != localId).collect(Collectors.toSet());
+        return voters.stream().filter(voterId -> voterId != localIdOrNil()).collect(Collectors.toSet());
+    }
+
+    public int localIdOrNil() {
```

Comment:
`localIdOrSentinel` would be more accurate, I think.

Reply:
Fair enough. I guess "nil" suggests zero.

Reply:
Yeah, that would be the closest for primitive types.